core/services/bulletprooftxmanager/bulletprooftxmanager.go

```diff
@@ -83,8 +85,9 @@ type TxManager interface {
 	httypes.HeadTrackable
 	service.Service
 	Trigger(addr common.Address)
-	CreateEthTransaction(db *gorm.DB, fromAddress, toAddress common.Address, payload []byte, gasLimit uint64, meta interface{}, strategy TxStrategy) (etx EthTx, err error)
+	CreateEthTransaction(db *gorm.DB, fromAddress, toAddress common.Address, payload []byte, gasLimit uint64, meta interface{}, strategy TxStrategy, taskRunID uuid.NullUUID, minConfirmations null.Uint32) (etx EthTx, err error)
 	GetGasEstimator() gas.Estimator
+	ResumeCallback(fn func(id uuid.UUID, value interface{}) error)
```

Comment:
This is almost there, but I think we can use a better approach.

How about "RegisterResumeCallback". This will call the same name function on EthConfirmer and add it to an array, all of which will be called.

Reply:
I was hoping to do this, but the `ec` isn't created until `Start` is called, and even then there's no reference to it from bpxtm: https://github.com/smartcontractkit/chainlink/blob/cfeb0f49b3b463a0572a6cc94f9719f0e84d9de2/core/services/bulletprooftxmanager/bulletprooftxmanager.go#L148-L149

Should I move that into the `New` function?

Reply:
Yeah makes more sense for the broadcaster and confirmer to be in New and fields on the bptxm, I don't see why they are special compared to the reaper/resender/gasestimator

Reply:
New seems like a more natural place for those, can't see why they are different than resender/gasestimator etc.

Reply:
i'd vote for calling it `RegisterResumeCallback` or `SetResumeCallback`